<a href="https://colab.research.google.com/github/anupstar100/Google-Play-Store-App-Review-EDA-/blob/main/Play_Store_App_Review_Analysis_Capstone_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> The Play Store apps data has enormous potential to drive app-making businesses to success. Actionable insights can be drawn for developers to work on and capture the Android market. </b>

## <b> Each app (row) has values for catergory, rating, size, and more. Another dataset contains customer reviews of the android apps.</b>

## <b> Explore and analyze the data to discover key factors responsible for app engagement and success. </b>

#Introduction

Google Play Store or formerly Android Market, is a digital distribution service developed and operated by Google. It is an official apps store that provides variety content such as apps, books, magazines, music, movies and television programs. It serves an as platform to allow users with 'Google certified' Android operating system devices to donwload applications developed and published on the platform either with a charge or free of cost. With the rapidly growth of Android devices and apps, it would be interesting to perform data analysis on the data to obtain valuable insights.

#Data Preparation and Cleaning

In [2]:
# IMPORTING NECESSARY LIBRARIES

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline
# ALLOW MATPLOTLIB TO PLOT INLINE WITH FRONTENDS

In [3]:
# MOUNT THE DRIVE

from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [28]:
# LOAD THE DATABASE

apps_df = pd.read_csv('/content/drive/MyDrive/Datasets/EDA/PlayStore/Copy of Play Store Data.csv')
apps_df.head()

App  ...   Android Ver
0     Photo Editor & Candy Camera & Grid & ScrapBook  ...  4.0.3 and up
1                                Coloring book moana  ...  4.0.3 and up
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ...  4.0.3 and up
3                              Sketch - Draw & Paint  ...    4.2 and up
4              Pixel Draw - Number Art Coloring Book  ...    4.4 and up

[5 rows x 13 columns]

In [ ]:
# LOOK AT THE RANDOM 10 RECORDS IN THE APPS DATAFRAME

apps_df.sample(10)

App  ...         Android Ver
7574            Inches to Centimeters  ...          1.6 and up
5692      Job'Of - Emploi au Cameroun  ...          4.0 and up
1863                      Idle Heroes  ...          2.3 and up
6922                    BW Map mobile  ...          4.0 and up
99     ipsy: Makeup, Beauty, and Tips  ...          4.1 and up
3999                        Learn C++  ...          4.0 and up
10684                 Gun Club Armory  ...          4.1 and up
9853     Amino: Communities and Chats  ...        4.0.3 and up
9790             ES Chromecast plugin  ...          4.0 and up
9417                     Sensenuts eI  ...  Varies with device

[10 rows x 13 columns]

# Unique Columns Data

In [ ]:
# UNIQUE CATEGORies

apps_df['Category'].unique()

array(['ART_AND_DESIGN', 'AUTO_AND_VEHICLES', 'BEAUTY',
       'BOOKS_AND_REFERENCE', 'BUSINESS', 'COMICS', 'COMMUNICATION',
       'DATING', 'EDUCATION', 'ENTERTAINMENT', 'EVENTS', 'FINANCE',
       'FOOD_AND_DRINK', 'HEALTH_AND_FITNESS', 'HOUSE_AND_HOME',
       'LIBRARIES_AND_DEMO', 'LIFESTYLE', 'GAME', 'FAMILY', 'MEDICAL',
       'SOCIAL', 'SHOPPING', 'PHOTOGRAPHY', 'SPORTS', 'TRAVEL_AND_LOCAL',
       'TOOLS', 'PERSONALIZATION', 'PRODUCTIVITY', 'PARENTING', 'WEATHER',
       'VIDEO_PLAYERS', 'NEWS_AND_MAGAZINES', 'MAPS_AND_NAVIGATION',
       '1.9'], dtype=object)

In [ ]:
# UNIQUE 'TYPE'

apps_df['Type'].unique()

array(['Free', 'Paid', nan, '0'], dtype=object)

In [ ]:
# UNIQUE RATINGS

apps_df['Rating'].unique()

array([ 4.1,  3.9,  4.7,  4.5,  4.3,  4.4,  3.8,  4.2,  4.6,  3.2,  4. ,
        nan,  4.8,  4.9,  3.6,  3.7,  3.3,  3.4,  3.5,  3.1,  5. ,  2.6,
        3. ,  1.9,  2.5,  2.8,  2.7,  1. ,  2.9,  2.3,  2.2,  1.7,  2. ,
        1.8,  2.4,  1.6,  2.1,  1.4,  1.5,  1.2, 19. ])

In [ ]:
# UNIQUE GENRES

apps_df['Genres'].unique()

array(['Art & Design', 'Art & Design;Pretend Play',
       'Art & Design;Creativity', 'Art & Design;Action & Adventure',
       'Auto & Vehicles', 'Beauty', 'Books & Reference', 'Business',
       'Comics', 'Comics;Creativity', 'Communication', 'Dating',
       'Education;Education', 'Education', 'Education;Creativity',
       'Education;Music & Video', 'Education;Action & Adventure',
       'Education;Pretend Play', 'Education;Brain Games', 'Entertainment',
       'Entertainment;Music & Video', 'Entertainment;Brain Games',
       'Entertainment;Creativity', 'Events', 'Finance', 'Food & Drink',
       'Health & Fitness', 'House & Home', 'Libraries & Demo',
       'Lifestyle', 'Lifestyle;Pretend Play',
       'Adventure;Action & Adventure', 'Arcade', 'Casual', 'Card',
       'Casual;Pretend Play', 'Action', 'Strategy', 'Puzzle', 'Sports',
       'Music', 'Word', 'Racing', 'Casual;Creativity',
       'Casual;Action & Adventure', 'Simulation', 'Adventure', 'Board',
       'Trivia', 'Role 

In [ ]:
# INFO OF DATAFRAME

apps_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


By diagnosing the DataFrame, we know that:

1.   There are 13 columns of properties with 10841 rows of data.
2.   Column 'Reviews', 'Size', 'Installs' and 'Price' are in the type of 'object'.
3.   Values of column 'Size' are strings representing size in 'M' as Megabytes, 'k' as kilobytes and also 'Varies with devices'.
4.   Values of column 'Installs' are strings representing install amount with symbols such as ',' and '+'.
5.   Values of column 'Price' are strings representing price with symbol '$'.
6.   There is only one float64 type column which is 'Rating'.

Hence, we will need to do some data cleaning.



#Data Cleaning

In [32]:
# 1. CLEAN THE REVIEWS DATA AND CHANGE THE TYPE FROM 'OBJECT' TO 'FLOAT'

reviews = [i for i in apps_df['Reviews']]

def clean_reviews(review_list):

  """
  An 'M has been found while changing the type,
  this function will replace the 'M' with million
  """

  cleaned_data = []
  for review in review_list:
    if 'M' in review:
      review = review.replace('M', '')
      review = float(review) * 1000000    # 1 M = 10,00,000
    cleaned_data.append(review)
  return cleaned_data

In [33]:
apps_df['Reviews'] = apps_df['Reviews'].astype(float)

ValueError: ignored

In [26]:
# 2. CLEAN SIZE DATA COLUMN AND CHANGE ITS TYPE FROM 'OBJECT' TO 'FLOAT'

sizes = [i for i in apps_df['Size']]

def clean_size(size_list):
  """ As sizes are repesented in 'M' or 'K', we
      remove 'M' and convert 'K' into 'M'"""
  cleaned_data = []
  for size in size_list:
    if 'M' in size:
      size = size.replace('M', '')
      size = float(size)
    elif 'K' in size:
      size = size.replace('k', '')
      size = float(size)
      size = size / 1024    # 1 MB = 1024 KB
    else:
      size = float(0)       # REPRESENTING 'VARIES WITH DEVICE' WITH VALUE 0
    cleaned_data.append(size)
  return cleaned_data

In [27]:
apps_df['Size'] = clean_size(sizes)
apps_df['Size'] = apps_df['Size'].astype(float)

TypeError: ignored

In [17]:
apps_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  float64
 4   Size            10841 non-null  float64
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(3), object(10)
memory usage: 1.1+ MB


In [19]:
apps_df.head(10)

App  ...   Android Ver
0     Photo Editor & Candy Camera & Grid & ScrapBook  ...  4.0.3 and up
1                                Coloring book moana  ...  4.0.3 and up
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ...  4.0.3 and up
3                              Sketch - Draw & Paint  ...    4.2 and up
4              Pixel Draw - Number Art Coloring Book  ...    4.4 and up
5                         Paper flowers instructions  ...    2.3 and up
6            Smoke Effect Photo Maker - Smoke Editor  ...  4.0.3 and up
7                                   Infinite Painter  ...    4.2 and up
8                               Garden Coloring Book  ...    3.0 and up
9                      Kids Paint Free - Drawing Fun  ...  4.0.3 and up

[10 rows x 13 columns]

In [24]:
apps_df.iloc[37]

App               Floor Plan Creator
Category              ART_AND_DESIGN
Rating                           4.1
Reviews                        36639
Size                               0
Installs                  5,000,000+
Type                            Free
Price                              0
Content Rating              Everyone
Genres                  Art & Design
Last Updated           July 14, 2018
Current Ver       Varies with device
Android Ver             2.3.3 and up
Name: 37, dtype: object


In [25]:
apps_df.iloc[58]

App               Restart Navigator
Category          AUTO_AND_VEHICLES
Rating                            4
Reviews                        1403
Size                              0
Installs                   100,000+
Type                           Free
Price                             0
Content Rating             Everyone
Genres              Auto & Vehicles
Last Updated        August 26, 2014
Current Ver                   1.0.1
Android Ver              2.2 and up
Name: 58, dtype: object